extensions/amp-analytics/0.1/test/test-events.js

```diff
@@ -385,6 +386,98 @@ describes.realWin('Events', {amp: 1}, env => {
     });
   });
 
+  describe('AmpStoryEventTracker', () => {
+    let tracker;
+
+    beforeEach(() => {
+      tracker = root.getTracker('story', AmpStoryEventTracker);
+    });
+
+    it('should initialize', () => {
+      expect(tracker.root).to.equal(root);
+    });
+
+    it('should listen on story events', () => {
+      const analyticsEvent = sandbox.stub();
+
+      const defaultStoryConfig = {
+        'on': 'story-page-visible',
+      };
+
+      const defaultVars = {
+        'storyPageIndex': '0',
+        'storyPageId': 'p4',
+        'storyPageCount': '4',
+      };
+
+      tracker.add(
+        analyticsElement,
+        'story-page-visible',
+        defaultStoryConfig,
+        analyticsEvent
+      );
+
+      const event = new Event('story-page-visible');
+      event.data = defaultVars;
+
+      tracker.sessionObservable_.fire(event);
```

Comment:
Re: using private properties during tests
Is there a way to dispatch the event rather than triggering the listener yourself?

Reply:
I tried triggering the event myself but it wasn't working. Let me know if you have any other ideas.

Reply:
I'm not very familiar with the analytics code, but the fact that the listener isn't receiving the event in the `should listen on story events` test is pretty worrying :(
Just wanted to point this out, maybe it's fine, I'll let you and the owners decide :)

Reply:
Not super familiar with the analytics code either, but I did see usage of some private properties in this file, which is why I thought it was fine. @zhouyx do you have any thoughts on this?

Reply:
Could you try `tracker.trigger()`?

Reply:
Hmm, the `AmpStoryEventTracker` doesn't have that method, am I missing something?

Reply:
oops, sorry i got confused with the customEventTracker. Nvm of the `trigger()` method. 

>I tried triggering the event myself but it wasn't working

Could you explain how you are triggering the event? My quick guess is the event is not dispatched from within the analytics root, but from the window.

Reply:
```
      const rootEl = root.getRootElement();
      rootEl.dispatchEvent(event);
      expect(analyticsEvent).to.be.calledOnce; // Not being called.
```

Tried different variations of this, let me know if I'm doing something wrong.

Reply:
Oh, it works using `root.getRoot()` but not with `root.getRootElement()` 🤔 I should've known better since the listener is installed with `root.getRoot()`, I just thought that `root.getRootElement()` would be in the same scope... my bad.